src/vs/workbench/contrib/debug/browser/debugHover.ts

```diff
@@ -178,30 +184,38 @@ export class DebugHoverWidget implements IContentWidget {
 		return this.domNode;
 	}
 
-	async showAt(position: Position, focus: boolean): Promise<void> {
+	async showAt(position: Position, focus: boolean): Promise<void | ShowDebugHoverResult> {
 		this.showCancellationSource?.cancel();
 		const cancellationSource = this.showCancellationSource = new CancellationTokenSource();
 		const session = this.debugService.getViewModel().focusedSession;
 
 		if (!session || !this.editor.hasModel()) {
 			this.hide();
-			return;
+			return ShowDebugHoverResult.NOT_AVAILABLE;
 		}
 
 		const result = await this.debugHoverComputer.compute(position, cancellationSource.token);
-		if (this.isVisible() && !result.rangeChanged) {
-			return;
+		if (cancellationSource.token.isCancellationRequested) {
+			this.hide();
+			return ShowDebugHoverResult.CANCELLED;
 		}
-
-		if (!result.range || cancellationSource.token.isCancellationRequested) {
+		if (!result.range) {
 			this.hide();
-			return;
+			return ShowDebugHoverResult.NOT_AVAILABLE;
+		}
+		if (this.isVisible() && !result.rangeChanged) {
+			this.hide();
```

Comment:
Previously, this case did not hide the hover, now you are hiding it.

Reply:
According to the current logic , `this.hide()` is relevant here isn't it ?

Reply:
```suggestion
```